symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java

```diff
@@ -0,0 +1,394 @@
+  private final UsersApi usersApi;
+  private final AuthSession authSession;
+
+  public UserService(UserApi userApi, UsersApi usersApi, AuthSession authSession) {
+    this.userApi = userApi;
+    this.usersApi = usersApi;
+    this.authSession = authSession;
+  }
+
+  /**
+   * Retrieve user details of a particular user.
+   * @param uid User Id
+   * @return Details of the user.
+   */
+  public V2UserDetail getUserDetailByUid(@NonNull Long uid) {
+    try {
+      return userApi.v2AdminUserUidGet(authSession.getSessionToken(), uid);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Retrieve a list of users in the company (pod) by a filter.
+   * If no filter is given, retrieving all users in the company (pod).
+   *
+   * @param filter using to filter users by
+   * @return List of retrieved users
+   */
+  public List<V2UserDetail> listUsersDetailByFilter(@Nullable UserFilter filter) {
+    try {
+      if (filter == null) {
+        return userApi.v2AdminUserListGet(authSession.getSessionToken(), null, null);
+      }
+      List<UserDetail> userDetailList = userApi.v1AdminUserFindPost(authSession.getSessionToken(), filter, null, null);
+      return userDetailList.stream().map(UserDetailMapper.INSTANCE::userDetailToV2UserDetail).collect(Collectors.toList());
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Add a role to an user.
+   *
+   * @param uid User Id
+   * @param roleId Role Id
+   */
+  public void addRoleToUser(@NonNull Long uid, @NonNull String roleId) {
+    try {
+      StringId stringId = new StringId().id(roleId);
+      userApi.v1AdminUserUidRolesAddPost(authSession.getSessionToken(), uid, stringId);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Remove a role from an user.
+   * @param uid User Id
+   * @param roleId Role Id
+   */
+  public void removeRoleFromUser(@NonNull Long uid, @NonNull String roleId) {
+    try {
+      StringId stringId = new StringId().id(roleId);
+      userApi.v1AdminUserUidRolesRemovePost(authSession.getSessionToken(), uid, stringId);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Get the url of avatar of an user
+   * @param uid User Id
+   * @return List of avatar urls of the user
+   */
+  public List<Avatar> getAvatarFromUser(@NonNull Long uid) {
+    try {
+      return userApi.v1AdminUserUidAvatarGet(authSession.getSessionToken(), uid);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Update avatar of an user
+   * @param uid User Id
+   * @param image The avatar image for the user profile picture.The image must be a base64-encoded.
+   */
+  public void updateAvatarOfUser(@NonNull Long uid, @NonNull String image) {
```

Comment:
Could we also provide prototypes that accept `byte[]` or `InputStream` for the image parameter?